function-images/tests/video-bench/object_recognition/recog.py

```diff
@@ -0,0 +1,174 @@
+# MIT License
+#
+# Copyright (c) 2021 Michal Baczun and EASE lab
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Azure Function to perform inference.
+"""
+from torchvision import transforms
+from PIL import Image
+import torch
+import torchvision.models as models
+
+import sys
+import os
+# adding python tracing sources to the system path
+sys.path.insert(0, os.getcwd() + '/../proto/')
+sys.path.insert(0, os.getcwd() + '/../../../../utils/tracing/python')
+import tracing
+import videoservice_pb2_grpc
+import videoservice_pb2
+
+import io
+import grpc
+import argparse
+import boto3
+
+from concurrent import futures
+from timeit import default_timer as now
+
+start_overall = now()
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-sp", "--sp", dest = "sp", default = "80", help="serve port")
+parser.add_argument("-zipkin", "--zipkin", dest = "url", default = "http://zipkin.istio-system.svc.cluster.local:9411/api/v2/spans", help="Zipkin endpoint url")
+
+args = parser.parse_args()
+
+# set aws credentials:
+AWS_ID = os.getenv('AWS_ACCESS_KEY', "")
+AWS_SECRET = os.getenv('AWS_SECRET_KEY', "")
+
+
+if tracing.IsTracingEnabled():
+    tracing.initTracer("recog", url=args.url)
+    tracing.grpcInstrumentServer()
+
+# Load model
+start_load_model = now()
+model = models.squeezenet1_1(pretrained=True)
+end_load_model = now()
+load_model_e2e = (end_load_model - start_load_model) * 1000
+print('Time to load model: %dms' % load_model_e2e)
+
+labels_fd = open('imagenet_labels.txt', 'r')
+labels = []
+for i in labels_fd:
+    labels.append(i)
+labels_fd.close()
+
+def processImage(bytes):
```

Comment:
remove timestamps, add spans instead

use 2 helper functions: `preprocess` and `infer`